resources/js/Pages/LoginPage.jsx

```diff
@@ -0,0 +1,31 @@
+import { Head, usePage } from '@inertiajs/react';
+import LoginFormPraktikan from '@/Components/ComponentsPraktikans/LoginFormPraktikan';
+import LoginFormAssistant from '@/Components/ComponentsAssistants/LoginFormAssistant';
+import Vector from '@/Components/ComponentsPraktikans/Vector';
+import ButtonGroup from '@/Components/ComponentsPraktikans/ButtonGroup';
+
+export default function LoginPage() {
+    const { url } = usePage();
+    const mode = new URLSearchParams(url.split('?')[1]).get('mode') || 'praktikan'; 
```

Comment:
should be doing it this way
```suggestion
    const mode = new URL(url).searchParams.get('mode') || 'praktikan'; 
```

never try to parse things manually if you dont know how to take care of all the edge cases, for example, your current parsing method will break here meanwhile the correct parsing should be on the bottom one of this example:
```bash
╭─f4r4w4y@goldenrock 
╰─$ node
Welcome to Node.js v18.16.0.
Type ".help" for more information.
> "https://test.com/??test".split("?")
[ 'https://test.com/', '', 'test' ]
> "https://test.com/??test".split("?")[1]
''
> new URL("https://test.com/??test").searchParams
URLSearchParams { '?test' => '' }
>
```

Reply:
okay, thank you for pointing this out and providing a clear explanation. I will update the code to use the URL API for parsing query parameters, as it is more reliable and handles edge cases better.

Reply:
may i know why would you need to build the fullUrl that way, is `url` not a full url?

Reply:
perhaps, if url is always guaranteed to be a full URL, building fullUrl might not be necessary. The reason for building it this way is to handle cases where url could be a relative path instead of a full URL. Sorry if i'm wrong

Reply:
this `url` comes from usePage() which in turns comes from the inertia middleware namely [`HandleInertiaRequests`](https://github.com/Zaidanfzn/daskomweb2/blob/main/app/Http/Middleware/HandleInertiaRequests.php), and as you can see youre actually already providing a url there in line 40 https://github.com/Zaidanfzn/daskomweb2/blob/ee5cfac79a52a8fa2d47837beebf39d009afd106/app/Http/Middleware/HandleInertiaRequests.php#L40, you can change it to this instead
```
                'location' => $request->fullUrl(),
```

and then use this location from the usePage() instead of `url` like so
```
const { ziggy } = usePage();
const mode = new URL(ziggy.location).searchParams.get('mode') || 'praktikan';
```

Reply:
i used useEffect in the LoginPage component because the mode state directly impacts the behavior of the ButtonGroup component, particularly for highlighting the currently active button. The useEffect ensures that the mode value is updated whenever the URL changes, allowing the button highlighting to update dynamically without needing a page reload.

Additionally, I’ve updated the ButtonGroup component to work seamlessly with the mode state. It handles URL updates using window.history.pushState, preventing unnecessary reloads and ensuring the current mode is reflected accurately.

I would appreciate any suggestions you might have to further refine this implementation. Here’s the relevant code:

LoginPage.jsx :
```
const { ziggy } = usePage();
    const [mode, setMode] = useState(
        ziggy?.location ? new URL(ziggy.location).searchParams.get('mode') || 'praktikan' : 'praktikan'
    );

    useEffect(() => {
        const updateMode = () => {
            const currentMode = new URL(window.location.href).searchParams.get('mode') || 'praktikan';
            console.log('Updated Mode:', currentMode);
            setMode(currentMode);
        };

        document.addEventListener('modeChange', updateMode);
        window.addEventListener('popstate', updateMode);

        return () => {
            document.removeEventListener('modeChange', updateMode);
            window.removeEventListener('popstate', updateMode);
        };
    }, []);
```


ButtonGroup.jsx :
```
<div className="flex gap-4">
      {buttonTypes.map(type => (
          <Link 
              key={type} 
              href={`/login?mode=${type}`} 
              onClick={(e) => {
                  e.preventDefault(); 
                  window.history.pushState({}, '', `/login?mode=${type}`); 
                  document.dispatchEvent(new Event('modeChange')); 
              }}
          >
              <ButtonMode 
                  type={type} 
                  isActive={currentMode === type} 
              />
          </Link>
      ))}
  </div>
```

Reply:
You dont have to do this manually 
```
onClick={(e) => {
                  e.preventDefault(); 
                  window.history.pushState({}, '', `/login?mode=${type}`); 
                  document.dispatchEvent(new Event('modeChange')); 
              }}
```
The Link component from inertia already does that internally, read the docs again

You might be adding this custom implementation to dispatch the custom event, but its not necessary also since you can just listen to the popState only instead of listening to both popState and modeChange which is the custom event

Reply:
So you can remove that custom onclick implementation and remove the addeventlistener on the modeChange custom event, thanks!